### PR TITLE
[coordinator] Configurable writes to leaving shards count towards consistency, add read level unstrict all

### PR DESCRIPTION
**Pull request by robskillington**
**What this PR does / why we need it**:

Allows for writes to leaving shards to count towards consistency and adds read consistency level unstrict all. Combined these settings allow for clusters under stress to perform a node join without removing the availability of the leaving node. Since this can cause issues where writes made to unavailable nodes count towards consistency the read unstrict all consistency level is made available so you can attempt always to try to read from all replicas. This means you won't optimistically short-circuit once reading from majority of nodes (with unstrict majority) and will always give the cluster a chance to read from all replicas (so if a write during a node join with write to a leaving shard that count towards consistency leaves a write on two nodes but the shard that is given up after leaving drops the data and the joining node didn't persist it, you are most likely to always read it from the single shard that still has the data).

Only advised to use this when the cluster is under stress, and also advised to be using repairs in this situation.

**Special notes for your reviewer**:

**Does this PR introduce a user-facing and/or backwards incompatible change?**:

```release-note
NONE
```

**Does this PR require updating code package or user-facing documentation?**:

```documentation-note
NONE
```

### CODE REVIEW

File: src/dbnode/client/config.go [hunk 1]
```diff
@@ -117,6 +117,10 @@ type Configuration struct {
 
 	// WriteShardsInitializing sets whether or not to write to nodes that are initializing.
 	WriteShardsInitializing *bool `yaml:"writeShardsInitializing"`
+
+	// ShardsLeavingCountTowardsConsistency sets whether or not to write to leaving shards
```
Comment (arnikola) [inline src/dbnode/client/config.go:121]: Nit: `whether or not writes to leaving shards count towards consistency, by default they so not`

Comment (rallen090) [inline src/dbnode/client/config.go:121]: nit: I think you can drop "to write to"?

File: src/dbnode/client/types.go [hunk 2]
```diff
@@ -419,6 +419,14 @@ type Options interface {
 	// initializing or not.
 	WriteShardsInitializing() bool
 
+	// SetShardsLeavingCountTowardsConsistency sets whether to count shards
+	// that are leaving or not towards consistency level calculations.
+	SetShardsLeavingCountTowardsConsistency(value bool) Options
+
+	// ShardsLeavingCountTowardsConsistency returns whether to count shards
+	// that are leaving or not towards consistency level calculations.
```
Comment (arnikola) [inline src/dbnode/client/types.go:427]: Nit `That are leaving to count for`

**Review by arnikola — APPROVED** [review]
Lgtm w nits

**Review by rallen090 — APPROVED** [review]
LGTM

**Pull request merged**